civic-tech-club/src/components/BudgetVis.js

```diff
@@ -0,0 +1,226 @@
+  Legend,
+  Filler,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+import zoomPlugin from 'chartjs-plugin-zoom'; // Import the zoom plugin
+import data from '../data/budget.json';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+
+// Register the required Chart.js components and the zoom plugin
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  Filler,
+  zoomPlugin // Register the zoom plugin here
+);
+
+// Helper to generate unique colors
+const COLORS = [
+  'rgba(75, 192, 192, 1)',
+  'rgba(255, 99, 132, 1)',
+  'rgba(54, 162, 235, 1)',
+  'rgba(255, 206, 86, 1)',
+  'rgba(153, 102, 255, 1)',
+  'rgba(255, 159, 64, 1)',
+];
+
+const BudgetVis = () => {
+  const [chartTitle, setChartTitle] = useState('');
+  const [chartData, setChartData] = useState({ labels: [], datasets: [] });
+  const chartRef = useRef(null); // Create a ref for the chart
+
+  useEffect(() => {
+    // This code will run once when the component loads
+    handleChartChange('Gross Operating Expenditures');
+  }, []); // The empty array makes it run only once, when the component mounts
+
+  const handleChartChange = (title) => {
+    setChartTitle(title); // Set chart title for debugging or displaying
+    const newChartData = getLine(title, data); // Pass title directly to getLine function
+    setChartData(newChartData); // Update chart data with the new dataset
+    resetZoom();
+  };
+
+  // Generate the line chart data with custom colors
+  function getLine(budgetIndex, data) {
+    const chartdata = {
+      labels: ['2024', '2025', '2026', '2027'], // Preset labels for the X-axis (years)
+      datasets: [],
+    };
+
+    data.forEach((service, index) => {
+      let proceed = false;
+      let budgetObject = {};
+
+      service.budget.forEach((element) => {
+        if (element['Budget Summary'] === budgetIndex) {
+          proceed = true;
+          budgetObject = element;
+        }
+      });
+
+      if (proceed) {
+        // Create a dataset for each service
+        const indivObject = {
+          label: service['service'], // Dataset label
+          data: [
+            parseBudgetValue(budgetObject['2024']),
+            parseBudgetValue(budgetObject['2025']),
+            parseBudgetValue(budgetObject['2026']),
+            parseBudgetValue(budgetObject['2027']),
+          ],
+          borderColor: COLORS[index % COLORS.length], // Assign a unique color
+          fill: false, // No fill for line chart
+          tension: 0.1, // Smooth the line
+        };
+        chartdata.datasets.push(indivObject);
+      }
+    });
+
+    return chartdata;
+  }
+
+  // Helper function to clean up budget values
+  function parseBudgetValue(value) {
+    if (value === 'N/A' || value === '-$0') return null;
+    if (typeof value === 'string') {
+      value = value.replace(/[\$,]/g, '').trim(); // Remove dollar signs, commas, and spaces
```

Comment:
we can use the browser's built-in Intl object to do this for us! It would be Intl.NumberFormatter of the top of my head, for reference.

Reply:
I wasn't able to find a way to do this easily - the data comes to us in the format $3,000. I was able to find a lot on how to format a pure number, but not much on how to remove formatting. Open to any suggestions

Reply:
ah gotcha. It would look something like this:
```
// Remove the currency symbol and commas
const numericValue = parseFloat(currencyString.replace(/[$,]/g, ''));

// Create a formatter
const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
});

// Format the currency
formatter.format(numericValue);
```

We can definitely address that later.